src/chain/kusama/api/events/identities.ts

```diff
@@ -1,64 +1,60 @@
-import {UnknownVersionError} from '../../../../utils'
-import {
-    IdentityIdentityClearedEvent,
-    IdentityIdentityKilledEvent,
-    IdentityIdentitySubRemovedEvent,
-    IdentityIdentitySubRevokedEvent,
-} from '../../types/events'
-import {ChainContext, Event} from '../../types/support'
+import { Event } from '../../../../processor';
+import { UnknownVersionError } from '../../../../utils'
+import { events } from '../../types'
 
 const IdentityCleared = {
-    decode(ctx: ChainContext, event: Event) {
-        let e = new IdentityIdentityClearedEvent(ctx, event)
-        if (e.isV1030) {
-            const [who, deposit] = e.asV1030
+    decode(event: Event) {
+        let e = events.identity.identityCleared;
+        if (e.v1030.is(event)) {
+            const [who, deposit] = e.v1030.decode(event);
             return {who, deposit}
-        } else if (e.isV9130) {
-            return e.asV9130
+        } else if (e.v9130.is(event)) {
+            return e.v9130.decode(event)
         } else {
             throw new UnknownVersionError(e)
         }
     },
 }
 
 const IdentityKilled = {
-    decode(ctx: ChainContext, event: Event) {
-        let e = new IdentityIdentityKilledEvent(ctx, event)
-        if (e.isV1030) {
-            const [who, deposit] = e.asV1030
+    decode(event: Event) {
+        let e = events.identity.identityKilled;
+        if (e.v1030.is(event)) {
+            const [who, deposit] = e.v1030.decode(event);
             return {who, deposit}
-        } else if (e.isV9130) {
-            return e.asV9130
+        } else if (e.v9130.is(event)) {
+            return e.v9130.decode(event)
         } else {
             throw new UnknownVersionError(e)
         }
     },
 }
 
 const IdentitySubRemoved = {
-    decode(ctx: ChainContext, event: Event) {
-        let e = new IdentityIdentitySubRemovedEvent(ctx, event)
-        if (e.isV2015) {
-            const [sub, main, deposit] = e.asV2015
-            return {sub, main, deposit}
-        } else if (e.isV9130) {
-            return e.asV9130
+    decode(event: Event) {
+        const identity = events.identity.subIdentityRemoved;
```

Comment:
The naming of these constants is inconsistent. The first couple of times it's just `e` and now it's `identity`.